pkg/ring/lifecycler.go

```diff
@@ -562,21 +565,13 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 		// but we need to update the local state accordingly.
 		i.setRegisteredAt(instanceDesc.GetRegisteredAt())
 
-		// If the ingester is in the JOINING state this means it crashed due to
-		// a failed token transfer or some other reason during startup. We want
-		// to set it back to PENDING in order to start the lifecycle from the
-		// beginning.
-		if instanceDesc.State == JOINING {
-			level.Warn(log.Logger).Log("msg", "instance found in ring as JOINING, setting to PENDING",
-				"ring", i.RingName)
+		// The instance may already be in the ring with one of the following states:
+		// - JOINING: this means it crashed due to a failed token transfer or some other reason during startup.
+		// - LEAVING: this means it crashed while shutting down or unregister from ring on shutdown is disabled.
+		// In both cases, we want to set it back to PENDING in order to start the lifecycle from the beginning.
+		if instanceDesc.State == JOINING || instanceDesc.State == LEAVING {
```

Comment:
In [this comment](https://github.com/cortexproject/cortex/pull/1921/files#r359214524) @pstibrany opines that PENDING is wrong because another leaving ingester may start to transfer tokens to this ingester.

I think I agree.  Can you clarify why, if we have a valid set of tokens, we would want to wait for a bit and then go active?